File: server/src/main/scala/producer/message_generator/data_generators/date_time/FixedDateTimeGenerator.scala
package producer.message_generator.data_generators.date_time

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import producer.message_generator.data_generators.json_generator.JsonGenerator

case class FixedDateTimeGenerator(
    unixEpochMillis: Long
)

object FixedDateTimeGenerator:
    def fromPb(v: pb.FixedDateTimeGenerator): FixedDateTimeGenerator =
        FixedDateTimeGenerator(
            unixEpochMillis = v.unixEpochMillis
        )

    def toPb(v: FixedDateTimeGenerator): pb.FixedDateTimeGenerator =
        pb.FixedDateTimeGenerator(
            unixEpochMillis = v.unixEpochMillis
        )

File: ui/components/ui/ProducerSession/message-generator/data-generators/date-time/fixed-date-time-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type FixedDateTimeGenerator = {
  type: 'fixed-date-time-generator',
  unixEpochMillis: number
};

export function fixedDateTimeGeneratorFromPb(v: pb.FixedDateTimeGenerator): FixedDateTimeGenerator {
  return {
    type: 'fixed-date-time-generator',
    unixEpochMillis: v.getUnixEpochMillis()
  };
}

export function fixedDateTimeGeneratorToPb(v: FixedDateTimeGenerator): pb.FixedDateTimeGenerator {
  const message = new pb.FixedDateTimeGenerator();
  message.setUnixEpochMillis(v.unixEpochMillis);

  return message;
}
